src/iterative/KOSMAPOSL/KOSMAPOSLReconstruction.cxx

```diff
@@ -205,20 +205,13 @@ post_processing()
   if (base_type::post_processing())
     return true;
   
-  this->subiteration_counter=0;
   this->anatomical_sd.resize(anatomical_image_filenames.size());
 
   if (this->anatomical_image_filenames.size()!=sigma_m.size()){
       error("The number of sigma_m parameters must be the same as the number of anatomical image filenames");
       return false;
   }
 
-
-  if(!this->only_2D){
-     this->num_elem_neighbourhood=this->num_neighbours*this->num_neighbours*this->num_neighbours ;}
-  else{
-     this->num_elem_neighbourhood=this->num_neighbours*this->num_neighbours ;
-      }
 for(int i = 0; i<=anatomical_image_filenames.size()-1; i++)
 {  if (!this->anatomical_image_filenames.empty()){
```

Comment:
This test doesn't make sense, as if `anatomical_image_filenames` is `empty()`, it won't go into this loop. Maybe you meant `anatomical_image_filenames[i].empty()`? I think this is what happened in https://github.com/SyneRBI/SIRF/issues/654. `set_defaults` sets it to `""`. It'll try to read that then, hence the error message.

Checking if a string is empty feels somewhat strange  to me, and possibly it'd be better to have `set_defaults` call `anatomical_image_filenames.clear()` (and also for other `vector`s). That seems to lead us somewhat further away though.

Reply:
shall I make the check on the anatomical prior sptr?

Reply:
like `is_null_ptr(this->anatomical_prior_sptr[0])`

Reply:
I feel that in `post_processing` a check on filenames is appropriate. `set_up` should check indeed on `is_null_ptr(this->anatomical_prior_sptr[i])`